packages/vscode-boxel-tools/src/file-system-provider.ts

```diff
@@ -57,12 +52,47 @@ export class Directory implements vscode.FileStat {
 
 export type Entry = File | Directory;
 
+const selectedRealmKey = 'selectedRealms';
 export class RealmFS implements vscode.FileSystemProvider {
   root = new Directory('');
   realmClients: Map<string, RealmAuthClient> = new Map();
   realmsInitialized = false;
   jwtPromises: Map<string, Promise<string>> = new Map();
 
+  private context: vscode.ExtensionContext;
+  private selectedRealms: Set<string> = new Set();
+  private realmDirNamePromises: Map<string, Promise<string>> = new Map();
+  private realmDirNameToUrl: Map<string, string> = new Map();
+
+  constructor(context: vscode.ExtensionContext) {
+    this.context = context;
+
+    let selectedRealms = this.context.globalState.get<string>(selectedRealmKey);
```

Comment:
Do we need to store this state? Does something happen if it's only in this.selectedRealms?

Reply:
Ah yes. Extensions are reactivated if the first folder in the first folder is changed in the workspace, so I think we typically get two activations.

We might be able to get away with a workspace configuration - we don't need it to be stored between windows right? That could also be related to the bug @backspace saw, but I'm not sure.

Reply:
I've added comments as we discussed [747b77f](https://github.com/cardstack/boxel/pull/1709/commits/747b77f4944b6195490c514c690e1b9c2883eeb6)